src/coordinate_systems/XYZT.jl

```diff
@@ -0,0 +1,174 @@
+"""
+
+    XYZT <: AbstractCoordinateSystem
+
+Cartesian coordinate system for four-vectors. Using this requires the implementation of the following interface functions:
+
+```julia
+LorentzVectorBase.x(::CustomFourVector)
+LorentzVectorBase.y(::CustomFourVector)
+LorentzVectorBase.z(::CustomFourVector)
+LorentzVectorBase.t(::CustomFourVector)
+```
+
+"""
+struct XYZT <: AbstractCoordinateSystem end
+coordinate_names(::XYZT) = (:x, :y, :z, :t)
+
+####
+# Interface functions
+####
+
+function t end
+
+function x end
+
+function y end
+
+function z end
+
+####
+# derived components
+####
+
+@inline function spatial_magnitude2(::XYZT, mom)
+  return x(mom)^2 + y(mom)^2 + z(mom)^2
+end
+
+@inline function spatial_magnitude(cs::XYZT, mom)
+  return sqrt(spatial_magnitude2(mom))
+end
+
+@inline function mass2(::XYZT, mom)
+  return t(mom)^2 - x(mom)^2 - y(mom)^2 - z(mom)^2
+end
+
+@inline function mass(::XYZT, mom)
+  m2 = mass2(mom)
+  if m2 < zero(m2)
+    # Think about including this waring, maybe optional with a global PRINT_WARINGS switch.
+    #@warn("The square of the invariant mass (m2=P*P) is negative. The value -sqrt(-m2) is returned.")
+    return -sqrt(-m2)
+  else
+    return sqrt(m2)
+  end
+end
+
+@inline function boost_beta(::XYZT, mom)
+  en = t(mom)
+  rho = spatial_magnitude(mom)
+  if !iszero(en)
+    rho / en
+  elseif iszero(rho)
+    return zero(rho)
+  else
+    throw(
+      ArgumentError(
+        "There is no beta for a four-vector with vanishing time/energy component"
```

Comment:
nit: Error messages should not be capitalized according to the Julia style guide.
```suggestion
        "there is no beta for a four-vector with vanishing time/energy component"
```